frontend/packages/console-app/src/components/namespace/NamespaceMenuToggle.tsx

```diff
@@ -0,0 +1,99 @@
+import * as React from 'react';
+import { MenuToggle, Popper } from '@patternfly/react-core';
+
+const NamespaceMenuToggle = (props: {
+  disabled: boolean;
+  menu: React.ReactElement;
+  isOpen: boolean;
+  shortCut?: string;
+  title: string;
+  onToggle: (state: boolean) => void;
+}) => {
+  const { menu, isOpen, shortCut, title, onToggle, disabled } = props;
+
+  const toggleRef = React.useRef(null);
+  const containerRef = React.useRef(null);
+
+  const inMenuItem = (event): boolean => {
+    let classList = event.target.classList ? [...event.target.classList] : [];
+    if (event?.target?.parentNode?.parentNode?.classList) {
+      // This is to grab items in the menu like the favorite star button applied by PatternFly
+      classList = classList.concat([...event.target.parentNode.parentNode.classList]);
+    }
+    return classList.some(
+      (c) =>
+        /pf-c-menu.*/.test(c) ||
+        /co-namespace-project-selector.*/.test(c) ||
+        /pf-c-switch__toggle.*/.test(c) ||
+        /pf-c-switch__input.*/.test(c),
+    );
+  };
```

Comment:
IMO, using CSS classes like this will be error-prone. We don't want keyboard nav to be dependent on hard-coded CSS class names. Instead, we should manage the propagation of events to make sure they only trigger when we want.

We can close the dropdown menu when a click event propagates to `window`. Then we just intercept click events with `stopPropagation` when we don't want to trigger that behavior (e.g. when clicking inside NamespaceMenu or NamespaceMenuToggle). The same thing applies to keyup events. We close the dropdown menu when a 'Tab' keyup event propagates to `window`, then block those events from propagating where appropriate. I fiddled around with this a bit and was able to get something working, so let me know if you have any questions about it.

Reply:
This was the pattern that was used either in the demo or in the select component in PF. They have since updated their approach to not use classnames, changed following PF's new pattern.